### PR TITLE
[BUILD] Bump maven download plugin from 1.8.1 to 1.11.3

### PR DESCRIPTION
**Pull request by pionCham**
# :mag: Description
## Describe Your Solution 🔧

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklist 📝

- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -228,7 +228,7 @@
         <jars.target.dir>${project.build.directory}/scala-${scala.binary.version}/jars</jars.target.dir>
 
         <maven.plugin.build.helper.version>3.3.0</maven.plugin.build.helper.version>
-        <maven.plugin.download.version>1.8.1</maven.plugin.download.version>
+        <maven.plugin.download.version>1.11.2</maven.plugin.download.version>
```
Comment (bowenliang123) [inline pom.xml:231]:
```suggestion
        <maven.plugin.download.version>1.11.3</maven.plugin.download.version>
```
1.11.3 has been just released, bring back the support for Java 8.
https://github.com/maven-download-plugin/maven-download-plugin/releases/tag/1.11.3

**Comment by bowenliang123** [comment]
`Error:  Failed to execute goal com.googlecode.maven-download-plugin:download-maven-plugin:1.11.3:wget (download-spark-release) on project kyuubi-download: Network is unreachable -> [Help 1]`
1.11.3 is not cached on Maven repo mirror. Let's rerun the CI tests later.

**Comment by pionCham** [comment]
> `Error: Failed to execute goal com.googlecode.maven-download-plugin:download-maven-plugin:1.11.3:wget (download-spark-release) on project kyuubi-download: Network is unreachable -> [Help 1]` 1.11.3 is not cached on Maven repo mirror. Let's rerun the CI tests later.

How can i rerun the CI tests manually without submitting a commit？

**Review by yaooqinn — APPROVED** [review]

**Review by bowenliang123 — COMMENTED** [review]
A late LGTM from me.

**Pull request closed without merging**